### DISCUSSION TITLE
How can I speedup with wsl2/ubuntu in Windows 11?

### DISCUSSION THREAD

**Discussion by hare0319** [JanDeDobbeleer/oh-my-posh#1316] [Troubleshoot] [answered]
Hi,

I setup oh-my-posh with my PowerShell in windows terminal, it works fine.
Invoking a normal `cd test` or `ls` takes only few ms.

But when I move to WSL2/Ubuntu, and setup oh-my-posh with same configuration
in bash. It normally takes an extra 200ms.
![image](https://user-images.githubusercontent.com/11740764/143526077-b7287d97-95bb-4328-af62-d883b00aad19.png)

Is there something wrong with my setting?

when I issue `oh-my-posh-wsl --debug --config nordForest.omp.json --shell bash` in bash, I got the following output
```shell
Here are the timings of segments in your prompt:

ConsoleTitle(false)  -   0 ms -
session(true)        -   0 ms - \[\]\[\]\[\] \[\]\[\]scheng\[\]\[\] \[\]\[\]\[\]
path(true)           -   0 ms - \[ransparentm\]\[\]\[\]  \[\]\[\]tmp\[\]\[\] \[\]\[\]\[\]
git(false)           -   0 ms -
executiontime(true)  -   0 ms - \[ransparentm\]\[\]\[\] \[\]\[\]ﮫ\[\]\[\] \[\]\[\]0ms\[\]\[\] \[\]\[\]\[\]
exit(false)          -   0 ms -
```
which takes around 450ms.

Is there any thing I can do to speed this up?

Thanks very much in advance

**Comment by JanDeDobbeleer** [comment, marked as answer]
Yes, remove oh-my-posh-wsl and install oh-my-posh natively on Linux with Homebrew (or move that executable to the Linux file system). WSL2 and access to the Windows file system is what makes everything slow.

**Reply by hare0319** [reply]
Cool, thanks very much
